lease/lessor.go

```diff
@@ -173,11 +168,17 @@ func (le *lessor) SetRangeDeleter(rd RangeDeleter) {
 // TODO: when lessor is under high load, it should give out lease
 // with longer TTL to reduce renew load.
 func (le *lessor) Grant(ttl int64) *Lease {
-	id := LeaseID(le.idgen.Next())
-
 	le.mu.Lock()
 	defer le.mu.Unlock()
 
+	// granted leases numbers are consistent via raft-pass through
+	// TODO: this bakes in the lease assignment values; upgrading
```

Comment:
This approach seems OK. I feel the alternative is slightly better though. I think it is not hard to implement either. We can add ID here https://github.com/coreos/etcd/blob/master/etcdserver/etcdserverpb/rpc.proto#L263 and generate a unique ID here https://github.com/coreos/etcd/blob/master/etcdserver/api/v3rpc/lease.go#L35 before we send the request through raft.

I think if we are going to do the alternative, we'd better do it now.